### DISCUSSION TITLE
[Help]: latest version stuck to 5.12.7 on qnap nas container station

### DISCUSSION THREAD

**Discussion by RobertoPaolinelli** [mbentley/docker-omada-controller#398] [Help] [answered]
### Controller Version

latest

### Describe Your Issue or Question

Hello everybody, I'm new on this world, so sorry for my question, but please I need your aid to understand...
I'm using qnap nas updated to last QTS version and container station 3.0.5 (maybe, anyway it's the last one too)
Two monhs ago I installed omada SW controller from conatiner station pointing to latest tag and I got 5.12.7
Today I'm trying to update the container after having stopped it, but release is always 5.12.7 while here I'm reading latest version is 5.13.xy

Can anyone please help me understanding what I'm wrong with?

Thanks a lot

Best Regards
Robertto

### Expected Behavior

I'm expecting omada controller updaed to 5.13

### Steps to Reproduce

Open Container Station on QNAP NAS QTS 5.1.4.2596 (2023/11/28)

Click on explore and type omada in the search box

![image](https://github.com/mbentley/docker-omada-controller/assets/2271886/11acc5a3-94a7-4996-8346-b32882fe814b)

![image](https://github.com/mbentley/docker-omada-controller/assets/2271886/d7626f7a-b959-4a0e-a850-281501dc6265)

### How You're Launching the Container

```plain
From QNAP nas container station
```

### Container Logs

```plain
2024/01/13 11:54:26
admin
Containers
Created container "omada-controller-sdn".
```

**Comment by mbentley** [comment]
First, I would stop using latest as the tag. It’s a bad practice that can lead to unexpected behavior. You should switch to the major.minor tagged images such as the 5.13 tag.

You need to pull an updated image in order the get updates.  Not sure how the app you’re using does that but that’s the concept. Pull the image, recreate the container and you are now running the new version.

**Comment by RobertoPaolinelli** [comment]
OK, thanks a lot for your very quick response.

Sorry for my questions but I'm really starting with containers...

Can you give a link to learn docker commands in case the app I'm using will
not allow me to pull the updated image?

Anyway the controller told me about the new version but after downloading
it I discovered it was a linux tar.gz archive: why that?

Thanks a lot
Roberto

**Reply by mbentley** [reply, marked as answer]
Pull the image:
`docker pull mbentley/omada-controller:5.13`

Stop the container (replace `omada-controller` with whatever the container name is):
`docker stop -t 60 omada-controller`

See [this](https://github.com/mbentley/docker-omada-controller#preventing-database-corruption) for understanding why we specify a stop timeout.

Remove the container:
`docker rm omada-controller`

Re-create the container using whatever method you want, such as [`docker run...`](https://github.com/mbentley/docker-omada-controller#example-usage) or [`docker compose up...`](https://github.com/mbentley/docker-omada-controller#using-docker-compose) or using whatever container launching tool you want.

So why the .tar.gz...
So the software inside the container has no idea that it is containerized so the controller software thinks it has been downloaded as a gzipped tarball, which is what the image originally uses to download the software so upgrading if you were not using a Docker image would be to download the tarball and replace the previous install.  Containers simplify that since the packaging all happens before and you just consume the image in a ready to go state.

**Comment by RobertoPaolinelli** [comment]
Thank you very much indeed for the time you have spent for me.
Best Regards
Roberto
